src/lavinmq/client/client.cr

```diff
@@ -143,6 +144,10 @@ module LavinMQ
           if @running
             process_frame(frame)
           else
+            if (RoughTime.monotonic - @closeok_timer) > 5.seconds
```

Comment:
this only works if the client still sends frames, but imagine a client that just consumes but doesn't properly send the CloseOk frame, eg. because it's busy with other things.. One possibility is to start a new fiber when we close the client, that sleeps 5s, checks if the connection is still open and then force closes it.

Reply:
The down side is of course that if we closed 1M connections, and thus spawn 1M new fibers, that can be very memory intensive..

Reply:
or we set a `read_timeout` on the socket when we try to gracefully close the client, so the read_loop will raise an TimeoutError if the client is silent.

Reply:
i think my last commit adds it in the way you're thinking, not sure how to test it easily